pywr-schema/src/timeseries/mod.rs

```diff
@@ -0,0 +1,153 @@
+mod polars_dataset;
+
+use ndarray::{Array1, Array2};
+use polars::error::PolarsError;
+use polars::prelude::DataType::Float64;
+use polars::prelude::{DataFrame, Float64Type, IndexOrder, Schema};
+use pywr_core::models::ModelDomain;
+use pywr_core::parameters::{Array1Parameter, Array2Parameter, ParameterIndex};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use thiserror::Error;
+
+use crate::{parameters::ParameterMeta, SchemaError};
+
+use self::polars_dataset::PolarsDataset;
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+#[serde(tag = "type")]
+enum TimeseriesProvider {
+    Pandas,
+    Polars(PolarsDataset),
+}
+
+struct ResamplingArgs {}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub struct Timeseries {
+    #[serde(flatten)]
+    names: ParameterMeta,
+    provider: TimeseriesProvider,
+    url: PathBuf,
+}
+
+impl Timeseries {
+    pub fn load(&self, domain: &ModelDomain, data_path: Option<&Path>) -> Result<DataFrame, SchemaError> {
+        match &self.provider {
+            TimeseriesProvider::Polars(dataset) => dataset.load(self.url.as_path(), data_path),
+            TimeseriesProvider::Pandas => todo!(),
+        }
+    }
+}
+
+fn align_and_resample(df: &DataFrame, domain: &ModelDomain) -> Result<DataFrame, SchemaError> {
+    // 1. check that time column contains datetime
+    // 2. get domain timestep frequency
+    // 3. get df timestep frequency
+    // 4. resample df if they don't match
+    // 5. slice df to align with domain start and end dates
+
+    todo!()
+}
+
+pub struct LoadedTimeseriesCollection {
+    timeseries: HashMap<String, DataFrame>,
+}
+
+impl LoadedTimeseriesCollection {
```

Comment:
This looks like good to me. So, basically the timeseries get copied into an Array parameter. This will be inefficient if there are multiple references to the same data. Each will create a new array in the model.  That's probably good enough for now. 

I think the only way to improve upon that would be replicate the idea of a timeseries in core. Perhaps this could be something like the array parameters. If they implemented `MultiValueParameter` it might allow you to create an `Array2Parameter` and then refer to specific columns?